<a href="https://colab.research.google.com/github/angelafrocha/Estatistica-com-Python-para-analise-de-dados/blob/main/ETLeAn%C3%A1liseEnem2019.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Projeto de Análise de Dados Enem 2019(SP)

Este projeto faz o ETL e uma pequena análise dos dados dos estudantes do estado de SP que se inscreveram no enem 2019 e tem o objetivo de fazer um tratamento dos dados e separar os treineiros dos candidatos classifocados

Os dados foram extraídos do site do INEP:

http://inep.gov.br/microdados

Dicionário dos dados: https://docs.google.com/spreadsheets/d/1kFtu-jIpROyygBfvAg9MiF9Nr40qcwMH/edit?usp=share_link&ouid=115783816334987757607&rtpof=true&sd=true

###Importação das Bibliotecas

In [54]:
import pandas as pd
import numpy as np

###Conexão com o Google Drive

In [55]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


###Extração dos dados

In [56]:
df = pd.read_csv('/content/drive/MyDrive/Datasets/microdados_enem_2019/microdados_enem_2019_sp.csv', sep=';', encoding='iso-8859-1')
#sep ; para o arq csv
#encoding iso-8859-1 = codificação dos caracteres para ler todas as especificidades da Língua Portuguesa

In [57]:
pd.set_option('display.max_columns',100)

In [58]:
df.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  CO_MUNICIPIO_NASCIMENTO  \
0            3                 1                1506500.0   
1            3                 1                1503804.0   
2            3                 1                1501402.0   
3            3                 1                3548708.0   
4            1                 1                3549409.0   

  NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  \
0    Santa Izabel do Pará              15.0               PA                1   
1                 Jacundá              15.0               PA                1   
2                   Belém              15.0               PA                2   
3   São Bernardo do Campo              35.0               SP                2   
4    São Joaquim da Barra              35.0               SP                1   

   TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  IN_TREINEIRO   CO_ESCOLA  \
0                4          1        1.0             0         NaN   
1                2          1        1.0             0         NaN   
2                0          3        1.0             0  15567761.0   
3                0          2        1.0             0  35904958.0   
4                3          1        1.0             0         NaN   

   CO_MUNICIPIO_ESC       NO_MUNICIPIO_ESC  CO_UF_ESC SG_UF_ESC  \
0               NaN                    NaN        NaN       NaN   
1               NaN                    NaN        NaN       NaN   
2         1501402.0                  Belém       15.0        PA   
3         3548708.0  São Bernardo do Campo       35.0        SP   
4               NaN                    NaN        NaN       NaN   

   TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  \
0                     NaN                 NaN              NaN   
1                     NaN                 NaN              NaN   
2                     4.0                 1.0              1.0   
3                     2.0                 1.0              1.0   
4                     NaN                 NaN              NaN   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               1               1               1               1      5646.0   
1               0               0               0               0         NaN   
2               1               1               1               1      5762.0   
3               1               1               1               1      4022.0   
4               0               0               0               0         NaN   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      5858.0      5929.0      6002.0          0                1.0   
1         NaN         NaN         NaN          1                NaN   
2      6411.0      6343.0      7318.0          0                1.0   
3      4248.0      5096.0      3932.0          1                1.0   
4         NaN         NaN         NaN          1                NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   

###Limpeza dos dados

In [59]:
df.shape
#percebe aqui que no estado de SP houveram 813.772 inscritos. Sendo de 15,97% de todo o Brasil que foi de 5095308 inscritos

(813772, 44)

In [60]:
df.dtypes
#análise dos tipos de atributos

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

In [61]:
df.head(3)

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   

   TP_COR_RACA  TP_NACIONALIDADE  CO_MUNICIPIO_NASCIMENTO  \
0            3                 1                1506500.0   
1            3                 1                1503804.0   
2            3                 1                1501402.0   

  NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  \
0    Santa Izabel do Pará              15.0               PA                1   
1                 Jacundá              15.0               PA                1   
2                   Belém              15.0               PA                2   

   TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  IN_TREINEIRO   CO_ESCOLA  \
0                4          1        1.0             0         NaN   
1                2          1        1.0             0         NaN   
2                0          3        1.0             0  15567761.0   

   CO_MUNICIPIO_ESC NO_MUNICIPIO_ESC  CO_UF_ESC SG_UF_ESC  \
0               NaN              NaN        NaN       NaN   
1               NaN              NaN        NaN       NaN   
2         1501402.0            Belém       15.0        PA   

   TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  \
0                     NaN                 NaN              NaN   
1                     NaN                 NaN              NaN   
2                     4.0                 1.0              1.0   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               1               1               1               1      5646.0   
1               0               0               0               0         NaN   
2               1               1               1               1      5762.0   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      5858.0      5929.0      6002.0          0                1.0   
1         NaN         NaN         NaN          1                NaN   
2      6411.0      6343.0      7318.0          0                1.0   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   
1            NaN            NaN            NaN            NaN            NaN   
2          160.0          120.0          200.0          200.0          200.0   

   NU_NOTA_REDACAO Q025  
0            940.0    B  
1              NaN    A  
2            880.0    A

In [62]:
#Removendo colunas que não serão analisadas
df1 = df.drop(columns=['CO_MUNICIPIO_RESIDENCIA',
                       'CO_UF_RESIDENCIA',
                       'CO_MUNICIPIO_NASCIMENTO',
                       'CO_UF_NASCIMENTO',
                       'SG_UF_RESIDENCIA',
                       'NO_MUNICIPIO_NASCIMENTO',
                       'CO_UF_NASCIMENTO',
                       'SG_UF_NASCIMENTO',
                       'TP_ANO_CONCLUIU',
                       'TP_ENSINO',
                       'CO_MUNICIPIO_ESC',
                       'CO_UF_ESC',
                       'SG_UF_ESC',
                       'TP_DEPENDENCIA_ADM_ESC',
                       'TP_LOCALIZACAO_ESC',
                       'TP_SIT_FUNC_ESC',
                       'CO_ESCOLA'], )

In [63]:
df1.head()
#foi observado aqui que as notas de CH,CN, Mat e Ling estão multilicadas

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0               1      5646.0      5858.0      5929.0      6002.0          0   
1               0         NaN         NaN         NaN         NaN          1   
2               1      5762.0      6411.0      6343.0      7318.0          0   
3               1      4022.0      4248.0      5096.0      3932.0          1   
4               0         NaN         NaN         NaN         NaN          1   

   TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0          160.0          200.0          180.0   
1                NaN            NaN            NaN            NaN   
2                1.0          160.0          120.0          200.0   
3                1.0           80.0          160.0          120.0   
4                NaN            NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          200.0          200.0            940.0    B  
1            NaN            NaN              NaN    A  
2          200.0          200.0            880.0    A  
3          100.0          100.0            560.0    B  
4            NaN            NaN              NaN    B

In [64]:
df1.shape

(813772, 28)

In [65]:
#Corrigindo as notas
df1.loc[:,'NU_NOTA_CN'] /= 10
df1.loc[:,'NU_NOTA_CH'] /= 10
df1.loc[:,'NU_NOTA_LC'] /= 10
df1.loc[:,'NU_NOTA_MT'] /= 10

In [66]:
df1.head(3)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   

  NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0              NaN               1               1               1   
1              NaN               0               0               0   
2            Belém               1               1               1   

   TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  \
0               1       564.6       585.8       592.9       600.2          0   
1               0         NaN         NaN         NaN         NaN          1   
2               1       576.2       641.1       634.3       731.8          0   

   TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0          160.0          200.0          180.0   
1                NaN            NaN            NaN            NaN   
2                1.0          160.0          120.0          200.0   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          200.0          200.0            940.0    B  
1            NaN            NaN              NaN    A  
2          200.0          200.0            880.0    A

In [67]:
#Renomeando colunas
df1 = df1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO',
                          'NU_NOTA_CN': 'NOTA_CN',
                          'NU_NOTA_CH': 'NOTA_CH',
                          'NU_NOTA_LC': 'NOTA_LC',
                          'NU_NOTA_MT': 'NOTA_MT',
                          'NU_NOTA_COMP1': 'COMP1',
                          'NU_NOTA_COMP2': 'COMP2',
                          'NU_NOTA_COMP3': 'COMP3',
                          'NU_NOTA_COMP4': 'COMP4',
                          'NU_NOTA_COMP5':'COMP5',
                          'NU_IDADE': 'IDADE',
                          'TP_SEXO': 'SEXO',
                          'TP_COR_RACA': 'RACA',
                          'Q025': 'INTERNET',
                          'TP_ESCOLA':'ESCOLA'})

In [68]:
df1.head(3)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL  RACA  \
0     1,90E+11            Santa Isabel     22    M                1     3   
1     1,90E+11             Pinhalzinho     19    M                1     3   
2     1,90E+11               São Paulo     17    F                1     3   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO NO_MUNICIPIO_ESC  \
0                 1                1       1             0              NaN   
1                 1                1       1             0              NaN   
2                 1                2       3             0            Belém   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NOTA_CN  \
0               1               1               1               1    564.6   
1               0               0               0               0      NaN   
2               1               1               1               1    576.2   

   NOTA_CH  NOTA_LC  NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  \
0    585.8    592.9    600.2          0                1.0  160.0  200.0   
1      NaN      NaN      NaN          1                NaN    NaN    NaN   
2    641.1    634.3    731.8          0                1.0  160.0  120.0   

   COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0  180.0  200.0  200.0         940.0        B  
1    NaN    NaN    NaN           NaN        A  
2  200.0  200.0  200.0         880.0        A

####Renomeando dados das colunas para melhor entendimento

In [69]:
df1['RACA'].unique()

array([3, 1, 4, 2, 0, 5])

In [70]:
df1["RACA"] = df1["RACA"].replace({0:"nao_declarado", 
                                   1:"branca", 
                                   2:"preta", 
                                   3:"parda", 
                                   4:"amarela", 
                                   5:"indigena"})

In [71]:
df1["TP_LINGUA"] = df1["TP_LINGUA"].replace({0:"ingles", 
                                             1:"espanhol"})

In [72]:
df1["ESCOLA"] = df1["ESCOLA"].replace({1:"nao_respondeu", 
                                       2:"publica", 
                                       3:"privada", 
                                       4:"exterior"})

In [73]:
df1["INTERNET"] = df1["INTERNET"].replace({"A":"nao", 
                                           "B":"sim"})

In [74]:
df1['TP_ESTADO_CIVIL'] = df1['TP_ESTADO_CIVIL'].replace({0:'nao_informado',
                                                         1:'solterio',
                                                         2:'casado/uniao_est',
                                                         3:'divorciado/separado',
                                                         4:'viuvo'})

In [75]:
df1['TP_NACIONALIDADE'] = df1['TP_NACIONALIDADE'].replace({0:'nao_informado',
                                                           1:'brasileiro',
                                                           2:'br_naturalizado',
                                                           3:'estrangeiro',
                                                           4:'br_nato_nasc_esterior'})

In [76]:
df1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO   TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M          solterio   parda   
1     1,90E+11             Pinhalzinho     19    M          solterio   parda   
2     1,90E+11               São Paulo     17    F          solterio   parda   
3     1,90E+11   São Bernardo do Campo     19    M          solterio   parda   
4     1,90E+11    São Joaquim da Barra     37    M  casado/uniao_est  branca   

  TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  \
0       brasileiro                1  nao_respondeu             0   
1       brasileiro                1  nao_respondeu             0   
2       brasileiro                2        privada             0   
3       brasileiro                2        publica             0   
4       brasileiro                1  nao_respondeu             0   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    ingles   
1               0      NaN      NaN      NaN      NaN  espanhol   
2               1    576.2    641.1    634.3    731.8    ingles   
3               1    402.2    424.8    509.6    393.2  espanhol   
4               0      NaN      NaN      NaN      NaN  espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      sim  
1                NaN    NaN    NaN    NaN    NaN    NaN           NaN      nao  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      nao  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      sim  
4                NaN    NaN    NaN    NaN    NaN    NaN           NaN      sim

###Análise Exploratória e Transformação dos dados

Análise das Idades

In [77]:
df1['IDADE'].value_counts()

17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
86         1
4          1
3          1
91         1
2          1
Name: IDADE, Length: 80, dtype: int64

In [78]:
#verificando os valores únicos
np.sort(pd.unique(df1['IDADE']))
#idades abaixo de 12 anos são suspeitas(alta probabilidade de erro)

array([ 2,  3,  4,  5,  6,  7,  8, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
       22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38,
       39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55,
       56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72,
       73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 86, 91])

In [79]:
df1.shape

(813772, 28)

In [80]:
#criando novo df apenas com >= 12 anos
df_maior11 = df1.loc[df1.IDADE > 11]

In [81]:
#Foram desconsideradas apenas 16 linhas, menos de 2%
df_maior11.shape

(813756, 28)

In [82]:
np.sort(pd.unique(df_maior11['IDADE']))

array([12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28,
       29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45,
       46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62,
       63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79,
       80, 81, 82, 86, 91])

#####Treineiros

In [83]:
#verificando a quantidade de treineiros
df_maior11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

1    93988
Name: IN_TREINEIRO, dtype: int64

In [84]:
#criando um df apenas com os dados dos treineiros
df_treineiros = df_maior11.loc[df_maior11.IN_TREINEIRO == 1]

In [85]:
df_treineiros.head(3)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO TP_ESTADO_CIVIL    RACA  \
20      1,90E+11                  Santos     15    F        solterio   parda   
21      1,90E+11                Panorama     16    F        solterio   parda   
106     1,90E+11               São Paulo     16    F        solterio  branca   

    TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  \
20        brasileiro                3  nao_respondeu             1   
21        brasileiro                3  nao_respondeu             1   
106       brasileiro                3  nao_respondeu             1   

    NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
20               NaN               1               1               1   
21               NaN               0               0               0   
106              NaN               1               1               1   

     TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
20                1    455.4    506.7    480.4    573.2    ingles   
21                0      NaN      NaN      NaN      NaN  espanhol   
106               1    424.5    517.7    489.6    532.6    ingles   

     TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO  \
20                 1.0  140.0  120.0  120.0  100.0   80.0         560.0   
21                 NaN    NaN    NaN    NaN    NaN    NaN           NaN   
106                1.0  120.0  120.0  120.0  100.0   80.0         540.0   

    INTERNET  
20       sim  
21       nao  
106      sim

In [86]:
#salvando df_treineiros para análise posterior
df_treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding = 'iso-8859-1', index = False)

#####Vestibulandos

In [87]:
# Separação entre os treineiros e os vestibulandos
df_vestibulandos = df_maior11.loc[df_maior11.IN_TREINEIRO == 0]

In [88]:
df_vestibulandos.shape

(719768, 28)

######Análise de Presença

In [89]:
df_vestibulandos['TP_PRESENCA_CN'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_CN, dtype: int64

In [90]:
df_vestibulandos['TP_PRESENCA_CH'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_CH, dtype: int64

In [91]:
df_vestibulandos['TP_PRESENCA_LC'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_LC, dtype: int64

In [92]:
df_vestibulandos['TP_PRESENCA_MT'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_MT, dtype: int64

**Resumo precença:**

Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.
Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.


Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.
Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

O par CH e LC foi aplicado no mesmo dia; o par CN e MAT foi aplicado no mesmo dia e posterior a CH+LC

In [93]:
df_vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: TP_STATUS_REDACAO, dtype: int64

######Valores Ausentes

In [94]:
df_vestibulandos['IDADE'].isnull().sum()

0

In [95]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [96]:
# Registros NAN nas notas
df_vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [97]:
#separando apenas os vestibulandos que foram classificados em todas as provas(nenhuma prova nula)
df_classificados = df_vestibulandos.loc[df_vestibulandos.TP_PRESENCA_CH == 1]
df_classificados = df_classificados.loc[df_classificados.TP_PRESENCA_MT == 1]

In [98]:
df_classificados.head(3)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO TP_ESTADO_CIVIL   RACA  \
0     1,90E+11            Santa Isabel     22    M        solterio  parda   
2     1,90E+11               São Paulo     17    F        solterio  parda   
3     1,90E+11   São Bernardo do Campo     19    M        solterio  parda   

  TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  \
0       brasileiro                1  nao_respondeu             0   
2       brasileiro                2        privada             0   
3       brasileiro                2        publica             0   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    ingles   
2               1    576.2    641.1    634.3    731.8    ingles   
3               1    402.2    424.8    509.6    393.2  espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      sim  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      nao  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      sim

In [99]:
df_classificados.shape

(499940, 28)

In [100]:
#Todos com treineiro = verdadeiro já foram reparados
df_classificados = df_classificados.drop(columns=['IN_TREINEIRO'])

In [101]:
df_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M        solterio   parda   
2     1,90E+11               São Paulo     17    F        solterio   parda   
3     1,90E+11   São Bernardo do Campo     19    M        solterio   parda   
5     1,90E+11              Embu-Guaçu     17    F        solterio   parda   
8     1,90E+11               Guarulhos     23    F        solterio  branca   

  TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA       NO_MUNICIPIO_ESC  \
0       brasileiro                1  nao_respondeu                    NaN   
2       brasileiro                2        privada                  Belém   
3       brasileiro                2        publica  São Bernardo do Campo   
5       brasileiro                2        publica                    NaN   
8       brasileiro                1  nao_respondeu                    NaN   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NOTA_CN  \
0               1               1               1               1    564.6   
2               1               1               1               1    576.2   
3               1               1               1               1    402.2   
5               1               1               1               1    399.6   
8               1               1               1               1    625.7   

   NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  \
0    585.8    592.9    600.2    ingles                1.0  160.0  200.0   
2    641.1    634.3    731.8    ingles                1.0  160.0  120.0   
3    424.8    509.6    393.2  espanhol                1.0   80.0  160.0   
5    426.8    487.2    419.2    ingles                1.0  120.0  120.0   
8    646.5    629.3    796.1    ingles                1.0  160.0  200.0   

   COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0  180.0  200.0  200.0         940.0      sim  
2  200.0  200.0  200.0         880.0      nao  
3  120.0  100.0  100.0         560.0      sim  
5  120.0  120.0   80.0         560.0      sim  
8  180.0  180.0  160.0         880.0      sim

In [102]:
df_classificados.shape

(499940, 27)

###Salvando(exportando) dataframe tratado

In [103]:
df_classificados.to_csv('enem_2019_tratado.csv', encoding = 'iso-8859-1', index = False)